metal/interrupt.h

```diff
@@ -80,6 +87,15 @@ struct metal_interrupt_vtable {
                            void *data);
     int (*mtimecmp_set)(struct metal_interrupt *controller, int hartid,
                         unsigned long long time);
+    int (*interrupt_affinity_enable)(struct metal_interrupt *controller,
+                                     metal_affinity bitmask, int id);
+    int (*interrupt_affinity_disable)(struct metal_interrupt *controller,
```

Comment:
If these methods return a `metal_affinity` then there can be fine-grained feedback instead of just an error code.

Reply:
OK, it is good to me. The relevant bit is 0 upon success, otherwise, it is 1.